File: src/backend/php/app/Models/EmailLog.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

/**
 * Eloquent model for the email_logs audit table.
 *
 * @property int              $id
 * @property string           $recipient
 * @property string           $subject
 * @property string           $body
 * @property string           $status       pending | sent | failed
 * @property \Carbon\Carbon   $created_at
 * @property \Carbon\Carbon|null $sent_at
 * @property string|null      $error_message
 */
class EmailLog extends Model
{
    /**
     * No Eloquent-managed updated_at column on this table.
     * created_at is handled by the DB default (useCurrent).
     */
    public $timestamps = false;

    protected $table = 'email_logs';

    protected $fillable = [
        'recipient',
        'subject',
        'body',
        'status',
        'sent_at',
        'error_message',
    ];

    protected $casts = [
        'created_at' => 'datetime',
        'sent_at'    => 'datetime',
    ];
}

File: src/backend/python/app/models/email_log.py
from datetime import datetime

from sqlalchemy import DateTime, Integer, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class EmailLog(Base):
    """Persists every outbound email attempt so the audit trail survives SMTP failures."""

    __tablename__ = "email_logs"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True, autoincrement=True)
    recipient: Mapped[str] = mapped_column(String(255), nullable=False)
    subject: Mapped[str] = mapped_column(String(500), nullable=False)
    body: Mapped[str] = mapped_column(Text, nullable=False)
    # pending | sent | failed
    status: Mapped[str] = mapped_column(String(50), nullable=False, default="pending")
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    sent_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)
    error_message: Mapped[str | None] = mapped_column(Text, nullable=True)

    def __repr__(self) -> str:
        return f"<EmailLog id={self.id!r} recipient={self.recipient!r} status={self.status!r}>"
